src/api/mod.rs

```diff
@@ -1390,6 +1413,78 @@ impl<'a> AuthenticatedApi<'a> {
     }
 }
 
+/// Available datasets for fetching organization events
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Dataset {
+    /// Our logs dataset
+    OurLogs,
+}
+
+impl Dataset {
+    /// Returns the string representation of the dataset
+    fn as_str(&self) -> &'static str {
+        match self {
+            Dataset::OurLogs => "ourlogs",
+        }
+    }
+}
+
+impl fmt::Display for Dataset {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+/// Options for fetching organization events
+pub struct FetchEventsOptions<'a> {
+    /// Dataset to fetch events from
+    pub dataset: Dataset,
+    /// Fields to include in the response
+    pub fields: &'a [&'a str],
+    /// Project ID to filter events by
+    pub project_id: Option<&'a str>,
+    /// Cursor for pagination
+    pub cursor: Option<&'a str>,
+    /// Query string to filter events
+    pub query: Option<&'a str>,
+    /// Number of events per page (default: 100)
+    pub per_page: Option<usize>,
+    /// Time period for stats (default: "1h")
+    pub stats_period: Option<&'a str>,
+    /// Sort order (default: "-timestamp")
+    pub sort: Option<&'a str>,
```

Comment:
For now, all of these fields appear to always be being set to a `Some` variant, so the default value never ends up being used. 

Therefore, I would like to see the struct fields be made non-optional, so we can also get rid of the defaults in the code, and make the API simpler. We can always later make the fields optional, as needed.
```suggestion
    /// Query string to filter events
    pub query: &'a str,
    /// Number of events per page
    pub per_page: usize,
    /// Time period for stats
    pub stats_period: &'a str,
    /// Sort order
    pub sort: &'a str,
```

Reply:
@szokeasaurusrex what are we trying to achieve with making things non-optional? Calls to the endpoint don't _need_ to have a query, stats_period, per_page or sort. What are we achieving by making them non-optional?

Reply:
I want to keep the Sentry CLI code simple. Making these optional means we have to add code paths to Sentry CLI, which for now, are not being used, because we always set the `query`, `stats_period`, `per_page`, and `sort`. Rust, however, still forces us to handle the cases of these being `None` (even though, in reality, they never are `None`). 

As a result, [we have code paths, which are completely unused](https://github.com/getsentry/sentry-cli/blob/74f859624e684896b1b94451ac765a846221778b/src/api/mod.rs#L1473-L1480), to handle the `None` cases. In the case of the `query`, we only provide it if it is there. For `stats_period`, `per_page`, and `sort`, we have default values defined, which are never used. From reading the code, however, it is not immediately clear that these values are not used.

The problem with keeping them as `Option` is just that it increases complexity and it makes it harder to see where the actual values get defined. I understand that the values are optional from the server perspective, but that absolutely does not mean that we need to make them optional in the CLI's API